libs/common/src/auth/services/token.service.ts

```diff
@@ -384,12 +391,16 @@ export class TokenService implements TokenServiceAbstraction {
         // Access token stored on disk unencrypted as platform does not support secure storage
         return await this.singleUserStateProvider
           .get(userId, ACCESS_TOKEN_DISK)
-          .update((_) => accessToken);
+          .update(() => accessToken, {
+            shouldUpdate: (previousValue) => previousValue !== accessToken,
+          });
       case TokenStorageLocation.Memory:
         // Access token stored in memory due to vault timeout settings
         return await this.singleUserStateProvider
           .get(userId, ACCESS_TOKEN_MEMORY)
-          .update((_) => accessToken);
+          .update(() => accessToken, {
```

Comment:
Not my ownership, but the linked epic describing the change clearly states that this applies to disk I/O only. Subsequently, there is no reason to apply it to memory state.
```suggestion
          .update((_) => accessToken);
```